src/blockloaders.js

```diff
@@ -478,23 +481,26 @@ class FileHandleLoader
 // ===========================================================================
 class IPFSRangeLoader
 {
-  constructor({url, headers}) {
+  constructor({url, headers, ...opts}) {
     this.url = url;
+    this.opts = opts;
 
     let inx = url.lastIndexOf("#");
     if (inx < 0) {
       inx = undefined;
     }
 
-    this.cid = this.url.slice("ipfs://".length, inx);
-
     this.headers = headers;
     this.length = null;
     this.canLoadOnDemand = true;
+  }
 
-    this.httpFallback = new FetchRangeLoader({url: "https://ipfs.io/ipfs/" + this.cid});
+  async initIPFS(opts=this.opts) {
+    const {api} = await createIPFS(opts);
```

Comment:
Should save this globally so that we don't call `createIPFS` on every fetch request?

Reply:
Added in latest version. 👍